docs/HowToFAQ/FAQRecursiveTermination.md

```diff
@@ -0,0 +1,49 @@
+---
+title: Is there a simple way to prove the termination of a recursive function?
+---
+
+## Question
+
+Is there a simple way to prove the termination of a recursive function?
+
+Here is an example function:
+```dafny
+datatype Dummy = State1 | State2
+
+  function WalkState(str: string, s: Dummy): string {
+    if str == [] then []
+    else if s.State1? then WalkState(str[1..], Dummy.State2)
+    else WalkState(str, Dummy.State1)
+  }
+```
+
+## Answer
+
+In general, to prove termination of any recursive structure one needs to declare a 
+(well-founded) measure that decreases on each iteration or recursive invocation;
+because a well-founded measure has a lower limit, the recursion must eventually terminate.
+In many cases, Dafny will deduce a satisfactory (provable) measure to apply by default.
+But where it cannot, the user must supply such a measure. A user-supplied measure is 
+declared in a `decreases` clause. Such a measure can be a tuple, ordered lexicographically by the
+termination metric for each data type.
```

Comment:
Can you please say something like "tuple without parentheses", because I've seen people not understanding why their `decreases (x, y)` won't work.

Lexicographic also means that longer sequences are "measurably smaller" than their prefixes. Can you please indicate that as well? An example is that a function with `decreases x` can call a function that `decreases x, 0` for the same value of `x`, that in turn calls a function that `decreases x` for `x-1`.

Reply:
Rather than put too much detail in the answer, I left the detailed description to the link to the RM.